ui/app/src/components/SingleFileUpload/SingleFileUpload.tsx

```diff
@@ -236,8 +251,8 @@ export function SingleFileUpload(props: SingleFileUploadProps) {
 
   useEffect(() => {
     const onUploadError = (file, error, response) => {
-      uppy.cancelAll();
```

Comment:
When in error, we used to cancel all uploads to avoid uppy triggering a 'complete' event (uppy triggers 'complete' even though there are errors). But cancelling the uploads caused the retry upload not to work. So, by removing this line and validating the successful files in the `onCompleteUpload` we avoid sending the completion action when there are errors, and the retry gets to work correctly.